packages/cli-repl/src/history.ts

```diff
@@ -0,0 +1,20 @@
+import redactInfo from 'mongodb-redact';
+
+/**
+ * Modifies command history array based on sensitive information.
+ * If redact option is passed, also redacts sensitive info.
+ *
+ * @param {array} History - Array of commands, where the first command is the
+ * most recent.
+ *
+ * @param {boolean} Redact - Option to redact sensitive info.
+ */
+function changeHistory(history: string[], redact: boolean = false) {
+  const hiddenCommands =
+    RegExp('createUser|auth|updateUser|changeUserPassword', 'ig');
+
+  if (hiddenCommands.test(history[0])) history.shift();
```

Comment:
Just found out a small one: you may want to return here. It does not cause direct issues, but in case of redact you are re-redacting the previous entry (if one). May throw if `history[0]` is `null` after shift and `redactInfo` is not happy about it.

Reply:
ah! good point, let me do a check if history actually has a thing.

in node repl this function only get triggered when the first item comes in, so it's always there, but might not be a case in other uses.

Reply:
The correct behaviour is to return, otherwise the previous element may be redacted twice (not sure that is a problem), maybe we need a specific test case for that if you don't have one.

Reply:
check the latest release! it's just returning ^__^